### PR TITLE
PLT: Order of plots does not preserve the column orders in df.hist

### LINKED ISSUE #29235: Categories reordered to alphabetical when plotting histograms
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

import matplotlib.pyplot as plt

df = pd.DataFrame({
    'width': [0.7, 0.2, 0.15, 0.2, 1.1],
    'length': [1.5, 0.5, 1.2, 0.9, 3]
    }, index= ['pig', 'rabbit', 'duck', 'chicken', 'horse'])

# Whether or not `column` is passed the order is still reordered to alphabetical.
df.hist(column=['width','length'])
plt.show()
```
#### Problem description

When plotting histograms from a DataFrame (see code-snippet above) the order of subplots is not maintained. Passing `column` does not appear to have any effect on the ordering either. 

#### Expected Output

Plots/subplots order equals column order in DataFrame.

#### Output of ``pd.show_versions()``

```
INSTALLED VERSIONS
------------------
commit: None
python: 3.7.3.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 69 Stepping 1, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.24.2
pytest: 5.2.1
pip: 19.1.1
setuptools: 41.0.1
Cython: None
numpy: 1.16.4
scipy: 1.2.2
pyarrow: None
xarray: 0.12.1
IPython: 7.5.0
sphinx: None
patsy: 0.5.1
dateutil: 2.8.0
pytz: 2019.1
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.1.0
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: None
bs4: 4.7.1
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10.1
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None
```

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [x] closes #29235
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by charlesdong1991** [comment]
could you please take a look? ^^  @WillAyd @jreback 

thanks!

File: doc/source/whatsnew/v1.1.0.rst [hunk 1]
```diff
@@ -536,6 +536,7 @@ Plotting
 - :func:`.plot` for line/bar now accepts color by dictonary (:issue:`8193`).
 - Bug in :meth:`DataFrame.plot.hist` where weights are not working for multiple columns (:issue:`33173`)
 - Bug in :meth:`DataFrame.boxplot` and :meth:`DataFrame.plot.boxplot` lost color attributes of ``medianprops``, ``whiskerprops``, ``capprops`` and ``medianprops`` (:issue:`30346`)
+- Bug in :meth:`DataFrame.hist` where the column is sorted given the ``column`` is assigned or not. (:issue:`29235`)
```
Comment (WillAyd) [inline doc/source/whatsnew/v1.1.0.rst:539]:
```suggestion
- Bug in :meth:`DataFrame.hist` where the order of the ``column`` argument was ignored (:issue:`29235`)
```

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for the ping. Minor nit on whatsnew but otherwise lgtm

I wasn't too familiar with `try_sort`. Looks like there's only one other use in pandas/core/indexes/api.py so separately might be worth cleaning up if not required

**Comment by charlesdong1991** [comment]
thanks for your review! @WillAyd I rephrase the whatsnew note

the build failure is unrelated

I will take a look at the other use of `try_sort` and see if it is needed, if not needed, I plan to have two follow-ups: one is to replace the `try_sort` in the other use, and then another one to deprecate the `try_sort`. Does it sound good to you? @WillAyd

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by charlesdong1991** [comment]
ping

seems it is the azure machine issue @WillAyd @jreback

**Comment by charlesdong1991** [comment]
ping @WillAyd @jreback

**Comment by WillAyd** [comment]
Great thanks @charlesdong1991

**Pull request merged**